@here/harp-map-theme/resources/berlin_tilezen_base.json

```diff
@@ -1,12 +1,15 @@
 {
     "sky": {
-        "type": "gradient",
-        "colorTop": "#015EBB",
-        "colorBottom": "#DDDCDA",
-        "groundColor": "#87959A",
-        "monomialPower": 1
+        "params": {
```

Comment:
Wouldn't it be simpler without this "params" indirection?
```
"sky": {
     "type": "gradient",
     "colorBottom": "#DDDCDA",
     ...
}
```

Reply:
The ``params`` indirection is handy at runtime, since it prevents us from checking for multiple attributes that ALL have to be defined for the sky to be generated accordingly.

The other possible ``Sky`` definition would be:
```
export interface Sky {
    type: "gradient" | "cubemap";
    topColor?: string;
    bottomColor?: string;
    groundColor?: string;
    monomialPower?: number;
    faces?: [
        CubemapFaceDefinition,
        CubemapFaceDefinition,
        CubemapFaceDefinition,
        CubemapFaceDefinition,
        CubemapFaceDefinition,
        CubemapFaceDefinition
    ];
}
```

In which only ``type`` would always be defined. Functions like ``isCubemapSky`` and ``isGradientSky`` would have to be used at runtime (which I'm not against if enough people agree this is the way to go).

Reply:
@amsgames94 I guess with this approach we would loose proper json schema. You could write sth. like:
`
"sky" : {
 "type": "cubemap",
 "topColor": "#fff"
`

Reply:
@amsgames94 
You could equally do this:
```
export type Sky = CubemapSkyParams | GradientSkyParams
``` 
abd have proper typing without artificial `params` node.

Moreover, not "renaming" (`colorTop => topColor`) for sake of renaming would help with backwards compatibility.

Now, we have to clean again (#510, #512) - not including, customers who will have to face with theme change.

Reply:
@zbigg The renaming didn't happen for the sake of renaming, it happened for the sake of consistency. We had ``colorBottom`` and ``colorTop`` and ``groundColor``, which had different naming approaches when they're all describing similar properties. Took the correct english grammar (adjective before noun) and applied to all members. I tried to fix all the themes, but might have slipped a few. But I still believe than we should try to clean old code instead of not touching it ever again... We can do better ;)

And regarding the ``Sky`` type, it's true that it could be simplified that way. Will look into it and create a minor.

Reply:
@zbigg Cleaner ``Sky`` definition based on your proposal on review now on #515